tests/tests.rs

```diff
@@ -61,48 +81,44 @@ impl EvmRpcSetup {
             env,
             caller,
             controller,
-            evm_rpc_id,
+            canister_id,
         }
     }
 
+    /// Shorthand for deriving an `EvmRpcSetup` with the caller as the canister controller.
     pub fn as_controller(&self) -> Self {
         let mut setup = self.clone();
```

Comment:
@rvanasa : understanding question, taking `&self` as a parameter to call afterwards `self.clone()` seems like an anti-pattern. Why not taking directly `self` as parameter and avoid cloning altogether? (similar question for the fluent methods below)

Reply:
Here is a motivating example:

```rs
let setup = EvmRpcSetup::new();

setup.as_controller().register_provider(...);
setup.as_anonymous().request(...);
```

Another possibility would be to require the caller to perform the cloning on their end:

```rs
...

setup.clone().as_controller().register_provider(...);
setup.clone().as_anonymous().request(...);
```

I see the case for switching to the latter if you think that would be preferable.

Reply:
@rvanasa : It's just that typically with the builder pattern you would do this in a single statement, e.g.,
```rust
let setupt = EvmRpcSetup::new()
  .as_controller().authorize(...)
  .register_provider(...)
  .as_anonymous().request(...);
```

Reply:
I'll do another pass on this while adding more tests. This would be the builder pattern equivalent:

```rust
EvmRpcSetup::new()
  .as_controller()
  .authorize(...)
  .as_user()
  .register_provider(...)
  .as_anonymous()
  .request(...);
```

The reason I'm using the current logic is that it's more explicit about which authorization to use. For instance, removing `as_user()` or `as_anonymous()` above could accidentally change the permissions for actions later in the chain. On the other hand, it's certainly more convenient, so I'll continue weighing both of these options.